timesync/peersync/peers.go

```diff
@@ -0,0 +1,70 @@
+package peersync
+
+import (
+	"context"
+
+	"github.com/spacemeshos/go-spacemesh/p2p/p2pcrypto"
+)
+
+type waitPeersReq struct {
+	ch  chan []p2pcrypto.PublicKey
+	min int
+}
+
+type peersWatcher struct {
+	requests chan *waitPeersReq
+}
+
+func (p *peersWatcher) waitPeers(ctx context.Context, n int) ([]p2pcrypto.PublicKey, error) {
```

Comment:
perhaps this code will be better located in p2p package as a utility for others to use as well

Reply:
makes sense. will move it

Reply:
i will open a tech debt issue for it and address it soon